src/main/java/com/google/api/codegen/util/java/JavaTypeTable.java

```diff
@@ -127,17 +131,21 @@ public static String getBoxedTypeName(String primitiveTypeName) {
   }
 
   @Override
-  public List<String> getImports() {
+  public Map<String, String> getImports() {
     // Clean up the imports.
-    List<String> cleanedImports = new ArrayList<>();
+    Map<String, String> cleanedImports = new TreeMap<>();
+    // Imported type is in java.lang or in package, can be ignored.
     for (String imported : imports.keySet()) {
       if (imported.startsWith(JAVA_LANG_TYPE_PREFIX)) {
-        // Imported type is in java.lang or in package, can be ignored.
         continue;
+      } else if (!implicitPackageName.isEmpty() && imported.startsWith(implicitPackageName)) {
+        // Imported type is in a subpackage must not be ignored.
+        if (!imported.substring(implicitPackageName.length() + 1).contains("\\.")) {
```

Comment:
Ok I feel bad for finding something else... but in any case, contains() takes an exact string, not a regex.